### ISSUE TITLE
/drive and +drives dont work together

### ISSUE THREAD

**Issue by ekorneechev** [FreeRDP/FreeRDP#4684]
If I use these keys together, only USB drives and gvfs are redirected. Home folder (or others)  not are redirected.

```
$ xfreerdp /v:server /u:user@domain /p:passwd -sec-nla /drive:home,/home/user +drives
[12:53:38:697] [23686:23687] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[12:53:38:697] [23686:23687] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[12:53:38:697] [23686:23687] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[12:53:38:935] [23686:23687] [ERROR][com.winpr.timezone] - Unable to get current timezone rule
[12:53:39:078] [23686:23687] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[12:53:39:078] [23686:23687] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[12:53:39:151] [23686:23687] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[12:53:39:153] [23686:23692] [INFO][com.freerdp.channels.rdpdr.client] - Loading device service drive [home] (static)
[12:53:39:175] [23686:23692] [INFO][com.freerdp.channels.rdpdr.client] - Loading device service drive [gvfs] (static)
[12:53:39:175] [23686:23692] [INFO][com.freerdp.channels.rdpdr.client] - Loading device service drive [2B74-8DEE] (static)
[12:53:41:323] [23686:23692] [INFO][com.freerdp.channels.rdpdr.client] - registered device #1: gvfs (type=8 id=2)
[12:53:41:324] [23686:23692] [INFO][com.freerdp.channels.rdpdr.client] - registered device #2: 2B74-8DEE (type=8 id=3)
[12:53:41:594] [23686:23687] [INFO][com.freerdp.client.x11] - Closed from X11

```

OS: Linux Mint 17.3
```
$ dpkg -s freerdp-x11 | grep Version
Version: 2.0.0~git201804050820-0+remmina201804082034.ra9ecd6a.d9be5820~ubuntu14.04.1
```
```
$ xfreerdp /buildconfig
This is FreeRDP version 2.0.0-dev2 (n/a)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=off HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=on WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DIRECTFB=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FFMPEG=ON WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=OFF WITH_ICU=OFF WITH_IPP=OFF WITH_JPEG=on WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PCSC=on WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_LEAK=OFF WITH_SANITIZE_LEAK_AVAILABLE=1 WITH_SERVER=OFF WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SSE2=ON WITH_THIRD_PARTY=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=OFF WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=OFF WITH_XRENDER=ON WITH_XSHM=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          None
CFLAGS:              -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wno-format-security -D_FORTIFY_SOURCE=2 -Wno-format-security  -march=i686 -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g
Compiler:            GNU, 4.8.4
Target architecture: x86
```

**Comment by ekorneechev** [comment]
The same problem occurs with the following command (home folder not are redirected.):
`$ xfreerdp /v:server /u:user@domain /p:passwd -sec-nla /drive:home,/home/user /drive:auto,*`

**Comment by speculzzz** [comment]
Hello...
If you place `/drive:auto,*` before the other `/drive:` parameters in the command line, then all will be ok:
`$ xfreerdp /v:server /u:user@domain /p:passwd -sec-nla /drive:auto,* /drive:home,/home/user`

**Comment by ekorneechev** [comment]
@speculzzz yes, but if i remove the USB flash drive from PC - folder 'home' also will removed.

**Comment by speculzzz** [comment]
@ekorneechev as I see the version [2.0.0-rc1](http://pub.freerdp.com/releases/freerdp-2.0.0-rc1.tar.gz) without this problem. I think the issue is inside the `first_hotplug()` function

**Comment by speculzzz** [comment]
@ekorneechev as I see the version [2.0.0-rc1](http://pub.freerdp.com/releases/freerdp-2.0.0-rc1.tar.gz) without this problem. I think the issue is inside the first_hotplug() function

**Comment by akallabeth** [comment]
The issue is that the automount/unmount will unmount all drives not part of mounted drives.
Needs a proper `do not touch` list for everything redirected by command line argument

**Comment by akallabeth** [comment]
Fixed by #4773
